aws_lambda_powertools/logging/formatter.py

```diff
@@ -148,16 +153,34 @@ def __init__(
         self.keys_combined = {**self._build_default_keys(), **kwargs}
         self.log_format.update(**self.keys_combined)
 
+        self.include_stacktrace = resolve_truthy_env_var_choice(env=os.getenv(constants.POWERTOOLS_STACKTRACE_ENV, "false"),
+                                                               choice=include_stacktrace,
+                                                               )
+
         super().__init__(datefmt=self.datefmt)
 
     def serialize(self, log: LogRecord) -> str:
         """Serialize structured log dict to JSON str"""
         return self.json_serializer(log)
+    
+    def serialize_traceback(self, e: Exception) -> list:
+        return [{"file": fs.filename, 
+                    "line": fs.lineno,
+                    "column": fs.colno,
+                    "function": fs.name,
+                    "statement": fs.line
+                    } for fs in traceback.extract_tb(e.__traceback__)]
```

Comment:
```suggestion
    def serialize_traceback(self, log_record: logging.LogRecord) -> list:
        print(log_record.exc_info[0], log_record.exc_info[1],log_record.exc_info[2])
        exception_info = {
            "type": log_record.exc_info[0].__name__,
            "value": str(log_record.exc_info[1]),
            "module": log_record.exc_info[1].__class__.__module__,
            "frames": []
        }

        for fs in traceback.extract_tb(log_record.exc_info[2]):
            frame_info = {
                "file": fs.filename,
                "line": fs.lineno,
                "function": fs.name,
                "statement": fs.line
            }
            exception_info["frames"].append(frame_info)

        return exception_info
```

If we assume that `serialize_traceback` always expects an exception, we break our current behavior where a customer can use `logger.exception("failed etc etc etc")`. We can extract the exception from the current `log_record` and parser it.

I left a [comment](https://github.com/aws-powertools/powertools-lambda-python/issues/1268#issuecomment-1743483745) in the issue talking about the data that we should include in the exception. I still don't have a strong opinion that this model I proposed is the best, so I would like to hear opinions.

Reply:
I updated the code per your suggestion.